vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,236 @@
+                allSpaceToSpacelinksListMap);
+
+        if (moduleList != null && !moduleList.isEmpty()) {
+            nodeFormation(request, moduleList, mapper, nodeArray, allNodeList, true);
+        }
+        sourceToTargetFormation(mapper, linkArray, allNodeList, spaceLinkMap);
+        model.addAttribute("overviewNode", mapper.writeValueAsString(nodeArray));
+        model.addAttribute("overviewLink", mapper.writeValueAsString(linkArray));
+        return "staff/spaces/graph";
+    }
+
+    /**
+     * To retrieve all space to space and space to module links corresponding to
+     * each space
+     * 
+     * @param request
+     * @param spaceList
+     * @param mapper
+     * @param nodeArray
+     * @param allNodeList
+     * @param spaceLinkMap
+     * @param allSpaceToModulelinksListMap
+     * @param allSpaceToSpacelinksListMap
+     */
+    private void getSpaceLinkMap(HttpServletRequest request, List<Space> spaceList, ObjectMapper mapper,
+            ArrayNode nodeArray, List<String> allNodeList, Map<String, List<String>> spaceLinkMap,
+            Map<String, List<IModuleLinkDisplay>> allSpaceToModulelinksListMap,
+            Map<String, List<ISpaceLinkDisplay>> allSpaceToSpacelinksListMap) {
+        if (spaceList != null && !spaceList.isEmpty()) {
+            nodeFormation(request, spaceList, mapper, nodeArray, allNodeList, false);
+            for (Space space : spaceList) {
+                List<IModuleLinkDisplay> spaceToModulelinksList = allSpaceToModulelinksListMap.get(space.getId());
+                List<ISpaceLinkDisplay> spaceLinks = null;
+                spaceLinks = allSpaceToSpacelinksListMap.get(space.getId());
+                List<ISpaceLinkDisplay> spaceToSpaceLinksList = null;
+                if (spaceLinks != null) {
+                    spaceToSpaceLinksList = new ArrayList<>();
+                    spaceToSpaceLinksList.addAll(spaceLinks);
+                }
+                List<String> spaceAllLink = new ArrayList<>();
+                if (spaceToModulelinksList != null && !spaceToModulelinksList.isEmpty()) {
+
+                    spaceToModulelinksList.forEach(link -> {
+                        spaceAllLink.add(link.getLink().getModule().getId());
+                    });
+                }
+                if (spaceToSpaceLinksList != null && !spaceToSpaceLinksList.isEmpty()) {
+
+                    spaceToSpaceLinksList.forEach(link -> {
+                        spaceAllLink.add(link.getLink().getTarget().getId());
+                    });
+                }
+                spaceLinkMap.put(space.getId(), spaceAllLink);
+            }
+        }
+    }
+
+    /**
+     * To map each space to all linked space or each space to all linked module
+     * 
+     * @param <T>
+     * @param spaceToAllOtherlinksList
+     * @param spaceToAllOtherlinksMap
+     * @param isModule
+     */
+    private <T> void getSpaceToAllOtherlink(List<T> spaceToAllOtherlinksList,
```

Comment:
no generics needed here; just pass `List<ILinkDisplay>` (I'm fairly certain that exists)

Reply:
also needs to be proper camelcase and plural `OtherLinks`

Reply:
After some modification of code the now is as follows:

Iterable<ModuleLinkDisplay> spaceToModuleLinks = moduleLinkDisplayRepository.findAll();
        Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap = new HashMap<>();
        getSpaceToAllOtherlink(spaceToModuleLinks, spaceToModuleLinksMap, true);

Iterable<SpaceLinkDisplay> spaceToSpaceLinksList = spaceLinkDisplayRepository.findAll();
        Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap = new HashMap<>();
        getSpaceToAllOtherlink(spaceToSpaceLinksList, spaceToSpaceLinksMap, false);

as the findAll is returing the class not the iterface. Moreover getLink() method is not present in ILinkDisplay so I think generic is useful here.Please suggest.

Reply:
your manager should not return a list of classes but a list of interfaces; besides that though leave it for now and I'll take another look when all changes are made.